process_report/tests/unit_tests.py

```diff
@@ -757,119 +757,179 @@ def test_apply_credit_error(self):
             test_invoice._get_pi_age(old_pi_df, "PI1", invoice_month)
 
 
-class TestBUSubsidy(TestCase):
-    def setUp(self):
-        data = {
-            "Invoice Month": [
-                "2024-03",
-                "2024-03",
-                "2024-03",
-                "2024-03",
-                "2024-03",
-                "2024-03",
-                "2024-03",
-                "2024-03",
-            ],
-            "Manager (PI)": ["PI1", "PI1", "PI2", "PI2", "PI3", "PI3", "PI4", "PI4"],
-            "Institution": [
-                "Boston University",
-                "Boston University",
+class TestBUSubsidyProcessor(TestCase):
+    def _assert_result_invoice(
+        self,
+        subsidy_amount,
+        test_invoice,
+        answer_invoice,
+        invoice_month="0000-00",
+    ):
+        new_bu_subsidy_proc = test_utils.new_bu_subsidy_processor(
+            invoice_month=invoice_month,
+            data=test_invoice,
+            subsidy_amount=subsidy_amount,
+        )
+        new_bu_subsidy_proc.process()
+        output_invoice = new_bu_subsidy_proc.data
+        answer_invoice = answer_invoice.astype(output_invoice.dtypes)
+        print(output_invoice)
+        print(answer_invoice)
+
+        self.assertTrue(output_invoice.equals(answer_invoice))
+
+    def _get_test_invoice(
+        self,
+        pi,
+        pi_balances,
+        balances=None,
+        project_names=None,
+        institution=None,
+        is_billable=None,
+        missing_pi=None,
+    ):
+        if not balances:
+            balances = pi_balances
+
+        if not project_names:
+            project_names = ["" for _ in range(len(pi))]
```

Comment:
If you don't care about the project name, I'd still give it a name so it's clear that it's not something you missed.

btw, this is also valid syntax if you just want repeat items in a list
```
>>> ['some-project']*len(pi)
['some-project', 'some-project', 'some-project', 'some-project']

```

Reply:
@QuanMPhm how we initialize the list was merely a suggestion, feel free to ignore it. However, I do like giving things some names instead of an empty string `""`.

Reply:
In the future, note that you can write instead:

```
project_names = [""] * len(pi)
```

No loops required!